BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/MainActivity.java

```diff
@@ -54,10 +69,40 @@ public int getIndicatorColor(int position) {
 
         tabs.setViewPager(pager);
 
+        // GeoFences
+
+        mGeofences = new ArrayList<Geofence>();
 
+        int size = BlocSpotApplication.getSharedDataSource().getAllPointsOfInterest().size();
 
+        for (int i = 0; i < size; i++) {
+            String title = pointsOfInterest.get(i).getTitle();
+            float lat = pointsOfInterest.get(i).getLatitude();
+            float lon = pointsOfInterest.get(i).getLongitude();
+
+            mGeofences.add(new Geofence.Builder()
+                    .setRequestId(title)
+                    .setCircularRegion(lat, lon, GEOFENCE_RADIUS)
+                    .setExpirationDuration(Geofence.NEVER_EXPIRE)
+                    .setLoiteringDelay(30000)
+                    .setTransitionTypes(
+                            Geofence.GEOFENCE_TRANSITION_ENTER
+//                                    | Geofence.GEOFENCE_TRANSITION_DWELL
+                                    | Geofence.GEOFENCE_TRANSITION_EXIT).build());
+        }
+
+        mGeofenceStore = new GeofenceStore(this, mGeofences);
     }
 
+    @Override
+    protected void onStart() {
```

Comment:
To be honest I don't think this was the problem with the notifications. If you don't provide additional behaviour to `onStart()` then you shouldn't override it (as it anyway the `super` is called). 

The same goes for `onStop()`.